src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+import {LitElement, html} from 'lit';
+import {customElement, state, property} from 'lit/decorators.js';
+import {animate_defs, animationCSS, ResponsiveVal,
+        getResponsiveValues, parseAttrVal} from "./utils";
+import {styleMap} from 'lit/directives/style-map.js';
+
+@customElement('aalam-tabs')
+export class AalamTabs extends LitElement {
+    DEFAULT_VALUES:{ [key:string]: any } = {
+        animationDur: 100,
+        activecls: 'tab-active',
+        fashion: 'xs:row',
+        colsize: 'title:30%;body:70%',
+        internal_fashion: 'row',
+        fashion_style: {xs:'row'},
+        column_size:{title:'30%', body:'70%'}
+    }
+
+    @property({type:String, attribute:true})
+    animation = '';
+
+    @property({type:Number, attribute:true})
+    animationDur = this.DEFAULT_VALUES.animationDur;
+
+    @property({type:String, attribute:true})
+    activecls = this.DEFAULT_VALUES.activecls;
+
+    @property({type:String, attribute:true})
+    fashion = this.DEFAULT_VALUES.fashion;
+
+    @property({type:String, attribute:true})
+    colsize = this.DEFAULT_VALUES.colsize;
+
+    @state()
+    _internal_fashion:string = this.DEFAULT_VALUES.internal_fashion;
+
+    private _prev_ix:number;
+    private _resizeListener = this._resizeEvent.bind(this);
+    private _animation_styles:{[key:string]:string} = {};
+    private _column_size:{[key:string]:string} =
+                this.DEFAULT_VALUES.column_size;
+    private _fashion_style:Array<ResponsiveVal> =
+                this.DEFAULT_VALUES.fashion_style;
+
+    constructor() {
+        super();
+    }
+    override attributeChangedCallback(
+                name:string, old_val:string, new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if(name == 'animation') {
+            if(animate_defs[new_val])
+                this._animation_styles = {open: new_val, close: new_val}
+            else {
+                this._animation_styles = parseAttrVal(new_val);
+                this._animation_styles = Object.assign({}, {
```

Comment:
Minor suggestion, instead of Object.assign we can just assign the second parameter to the left hand side, give the fact that the first parameter is empty.